tests/utils/test_variable_name_utils.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests Variable Name"""
+from unittest import TestCase
+
+from o2a.utils.variable_name_utils import convert_to_python_variable
+
+
+# pylint: disable=invalid-name
+class convert_to_python_variableTestCase(TestCase):
+    def test_should_keep_original_when_its_not_required(self):
+        self.assertEqual(convert_to_python_variable("TEST"), "TEST")
+
+    def test_should_replace_invalid_characters_to_underscore(self):
```

Comment:
It would be great to have parameterized here as well

Reply:
Is there any benefit from this? This is a pure function. This is a pure function, so testing this function does not cause any problems. Parameterized weakly integrates with IDE.

Reply:
I will prepare a change to be able to talk about two solutions

Reply:
> The KISS principle states that most systems work best if they are kept simple rather than made complicated; therefore, simplicity should be a key goal in design, and unnecessary complexity should be avoided.

Parametrized will not bring benefits here, so it's better to keep it simple

Reply:
We already use parameterized in similar tests. And there is a clear benefit - in case one of parameterized tests fails - the other are still executed. If you have series of asserts in one test, if first fails, the rest will not be executed. That might lead to unnecessary re-running the tests assert-by-assert. I think it should be our default behaviour that in case we have same tests with only input/output differing, we should use parameterised.